restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -673,12 +669,11 @@ public JsonNode searchDocumentsV2(
         List<Row> docRows = entry.getValue();
         for (Row row : docRows) {
           JsonNode jsonDoc =
-              getJsonConverterService()
-                  .convertToJsonDoc(
-                      Collections.singletonList(row),
-                      new DeadLeafCollectorNoOp(),
-                      true,
-                      db.treatBooleansAsNumeric());
+              jsonConverterService.convertToJsonDoc(
+                  Collections.singletonList(row),
+                  new DeadLeafCollectorNoOp(),
```

Comment:
Instead of creating a new `DeadLeafCollectorNoOp` each time, you could just create it once and use it everywhere. 🙂 
It would be nice if this was "non instantiable" that could be reused across the code. My suggestions:

1) Make it a `enum` as below and call it `DeadLeafCollectorNoOp.INSTANCE`
```
public enum DeadLeafCollectorNoOp implements DeadLeafCollector {
     INSTANCE;
     (...)
 ```
 
 2) Make it a singleton  non instantiable class as below:
 ```
 public final class DeadLeafCollectorNoOp implements DeadLeafCollector {
     private DeadLeafCollectorNoOp() {}
     
     private static final DeadLeafCollectorNoOp INSTANCE = new DeadLeafCollectorNoOp();
     
     public static DeadLeafCollectorNoOp getInstance() {
                return INSTANCE;
     }
     (...)
 ```

Reply:
I'd also recommend to have a second signature to this method that doesn't have the DeadLeafCollector for the cases it's not needed.
```
public JsonNode convertToJsonDoc(List<Row> rows, boolean writeAllPathsAsObjects, boolean numericBooleans) {
       return convertToJsonDoc(rows, DeadLeafCollector.INSTANCE, writeAllPathsAsObjects, numericBooleans);
}
```

Reply:
`@Value.Immutable()` && `@Value.Style(Singleton = true)` is also good option for singletons?